oxrdflib/parser.py

```diff
@@ -0,0 +1,121 @@
+import warnings
+from typing import Any, Optional, Union
+
+from rdflib import ConjunctiveGraph, Graph
+from rdflib.exceptions import ParserError
+from rdflib.parser import FileInputSource, InputSource, Parser
+
+from oxrdflib._converter import to_ox
+from oxrdflib.store import OxigraphStore
+
+__all__ = [
+    "OxigraphTurtleParser",
+    "OxigraphNTriplesParser",
+    "OxigraphRdfXmlParser",
+]
+
+
+class OxigraphParser(Parser):
+    def __init__(self):
+        pass
+
+    def parse(
+        self,
+        source: InputSource,
+        sink: Union[Graph, ConjunctiveGraph],
+        format: str,
+        encoding: Optional[str] = "utf-8",
+        **kwargs: Any,
+    ) -> None:
+        if encoding not in (None, "utf-8"):
+            raise ParserError("N3/Turtle files are always utf-8 encoded, I was passed: %s" % encoding)
+
+        if type(source) not in (FileInputSource, InputSource):
+            raise ParserError("Source must be either io(bytes) or io(str) or str or pathlib.Path")
+
+        if not isinstance(sink.store, OxigraphStore):
+            warnings.warn(
+                "Graph store should be an instance of OxigraphStore, "
+                f"got {type(sink.store).__name__} store instead."
+                " Attempting to parse using rdflib native parser.",
+                stacklevel=2,
+            )
+            sink.parse(source, format=format)
+
+        else:
+            base_iri = sink.absolutize(source.getPublicId() or source.getSystemId() or "")
+
+            if kwargs.get("transactional", False):
```

Comment:
I would maybe make `transactional` the default: having all methods on `Store` being atomic is a nice property. WDYT?

```suggestion
            if kwargs.get("transactional", True):
```

Reply:
That makes sense as we indeed should have same behaviour as rdfnative parsers (which are transactional)